Cherry-picks for 0.3.0

XVilka opened the pull request:
<!-- Filling this template is mandatory -->

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

<!-- Explain the **details** for making this change. Is a new feature implemented? What existing problem does the pull request solve? How does the pull request solve these issues? Please provide enough information so that others can review your pull request. -->

...

**Test plan**

<!-- What steps should the reviewer take to test your pull request? Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots/videos. This is your time to re-check that everything works and that you covered all the edge cases -->

...

**Closing issues**

<!-- put "closes #XXXX" in your comment to auto-close the issue that your PR fixes (if any). -->

...

kazarmy committed:
Use UTC timezone for tests

ret2libc committed:
Fix function pointers printing in complex types

wargio committed:
Resolve MMIO and extended registers (#1631)

ret2libc committed:
Update tree-sitter to 0.19.5 (#1633)

XVilka committed:
Fix enum printing

XVilka committed:
Fix array printing in complex types

officialcjunior committed:
Use PJ in `config_print_node()` (#1488)

* Remove `cb_printf` from `RzConfig`
* Rename `rz_config_list` to `rz_core_config_print_all`
* Remove `rz_cmd_eval` and run clang-format

wargio commented:
```
librz_asm.a(arch_arc_gnu_arc-dis.c.obj) : error LNK2005: SocketNotificationRetrieveEvents already defined in librz_analysis.a(.._asm_arch_xtensa_gnu_xtensa-isa.c.obj)
librz_asm.a(arch_arc_gnu_arcompact-dis.c.obj) : error LNK2005: SocketNotificationRetrieveEvents already defined in librz_analysis.a(.._asm_arch_xtensa_gnu_xtensa-isa.c.obj)
librz_asm.a(arch_hppa_gnu_hppa-dis.c.obj) : error LNK2005: SocketNotificationRetrieveEvents already defined in librz_analysis.a(.._asm_arch_xtensa_gnu_xtensa-isa.c.obj)
librz_asm.a(arch_xtensa_gnu_xtensa-dis.c.obj) : error LNK2005: SocketNotificationRetrieveEvents already defined in librz_analysis.a(.._asm_arch_xtensa_gnu_xtensa-isa.c.obj)
```

XVilka committed:
Add tests for `rz_type*_as_string()` functions

XVilka committed:
Fix Windows functions test

XVilka committed:
Fixing Coverity detected leaks in types (#1644)

* Fix CID 356334
* Fix CID 356331
* Fix CID 356330

ret2libc committed:
core: While doing @dm gather all addresses first, then iterate

This way if a command executed in the loop changes the dbg->maps, we
won't access unavailable memory with the next iteration of the loop.

This was happening when doing `\a jmp rsp @@dm` because of
`rz_debug_map_sync` which replace the dbg->maps value.

wargio committed:
Default build to Windows 10/Windows Server 2016 (#1649)

* LNK2005: SocketNotificationRetrieveEvents already defined
* use NTDDI_WIN10

wargio committed:
Remove yank test where we alloc 2 GB of ram for no reason (#1652)

Yank allocs 1GB via malloc(size) and then allocs another GB via RzBuffer

ret2libc committed:
Revert "Do not glitch the prompt when using scr.color=0 ##cons (#18057)" (#1647)

This reverts commit c90aee3c85906f45a46d65db7f657e3e5c7f3ba8.

Signed-off-by: Riccardo Schirone <sirmy15@gmail.com>
